services/base.js

```diff
@@ -175,13 +176,41 @@ class BaseService {
     )
   }
 
+  static get _regexFromPath() {
+    const { pattern } = this.url
+    const fullPattern = `${this._makeFullUrl(
+      pattern
+    )}.:ext(svg|png|gif|jpg|json)`
+
+    const keys = []
+    const regex = pathToRegexp(fullPattern, keys, {
+      strict: true,
+      sensitive: true,
+    })
+    const capture = keys.map(item => item.name).slice(0, -1)
+
+    return { regex, capture }
+  }
+
   static get _regex() {
-    // Regular expressions treat "/" specially, so we need to escape them
-    const escapedPath = this.url.format.replace(/\//g, '\\/')
-    const fullRegex = `^${this._makeFullUrl(
-      escapedPath
-    )}.(svg|png|gif|jpg|json)$`
-    return new RegExp(fullRegex)
+    const { pattern, format, capture } = this.url
+    if (pattern) {
+      if (format || capture) {
+        throw Error(
```

Comment:
I know we didn't have this check before, but can we also add check here that if we _haven't_ got a `pattern` key we should have both a `format` and `pattern` and throw a nice error if we haven't. At the moment we'll get something like `TypeError: Cannot read property 'replace' of undefined` or `Service declares incorrect number of capture groups`

Reply:
I think you meant `capture`, not `pattern`, though `capture` isn't required (and there's a test that it works with no `capture`).

But yea, I believe `format` is. I'll add a check for that.

Reply:
Yeah, sorry.. wordsing is hard. What you said :)